src/chase/impl/basic.rs

```diff
@@ -103,7 +103,7 @@ impl Model {
     fn record(&mut self, witness: WitnessTerm) -> E {
         match witness {
             WitnessTerm::Elem { element } => {
-                if self.domain().contains(&&element) {
+                if let Some(_) = self.domain().iter().find(|e| element.eq(e)) {
```

Comment:
Is this just:

```suggestion
                if let Some(_) = self.domain().iter().find(|e| element == e) {
```

Reply:
that'd give me a type mismatch: no implementation for `&&&chase::E == chase::E`